src/demo/ContextMenuPopup.tsx

```diff
@@ -0,0 +1,113 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import {ContextMenuItemType, ContextMenu, ContextMenuMessage} from './components/common/ContextMenu';
+
+import '../../res/demo/css/w3.css';
+
+
+const uuid: string = fin.desktop.Application.getCurrent().uuid;
+
+/**
+ * Simplification of ContextMenuItem - contains just the information required by the UI.
+ */
+export interface MenuItem {
+    id: string;
+    type: ContextMenuItemType;
+    caption: string;
+    hasChildren: boolean;
+}
+
+interface ContextMenuPopupProps {
+    items?: MenuItem[];
+    children?: React.ReactNode;
+}
+
+// tslint:disable-next-line:variable-name
+export const ContextMenuPopup: React.FunctionComponent<ContextMenuPopupProps> = (props) => {
+    const [menuItems, setMenuItems] = React.useState<MenuItem[]>([]);
+    let windowHeight = 2;
+
+    React.useEffect(() => {
+        const handleMenuMessage = (message: ContextMenuMessage) => {
+            if (message.action === "update") {
+                setMenuItems(message.items);
+                const ofWindow: fin.OpenFinWindow = fin.desktop.Window.getCurrent();
+                ofWindow.isShowing((showing) => {

```

Comment:
We should look to switch over any v1 API calls to v2

Reply:
I'll change calls to the v1 API in the other files but leave the ContextMenu as they currently are.  I will replace the context menu with a different one I'm writing based on the POC you sent me.

Reply:
Where's the other PoC?